### DISCUSSION TITLE
Prefer `fmt.Errorf(...)` to `fmt.Errorf(fmt.Sprintf(...))`

### DISCUSSION THREAD

**Discussion by cdignam-segment** [dominikh/go-tools#1292] [Ideas]
This is a style preference, but instead of writing `fmt.Errorf(fmt.Sprintf("failed to connect to db: %w", err))`, a developer can write `fmt.Errorf("failed to connect to db: %w", err)`.

**Comment by dominikh** [comment]
`fmt.Errorf(fmt.Sprintf("failed to connect to db: %w", err))` is plain wrong and will produce incorrect results if the string returned by `fmt.Sprintf` contains percent signs. I'm not inclined to offer a "simplification" here when the original code is wrong. But we might consider flagging the bug.
